extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilder.scala

```diff
@@ -0,0 +1,496 @@
+
+      case "AlterTableChangeColumnCommand" =>
+        val table = getTableName
+        val cols = getPlanField[String]("columnName") :: Nil
+        outputObjs += tablePrivileges(table, cols)
+
+      case "AlterTableDropPartitionCommand" =>
+        val table = getTableName
+        val cols = getPlanField[Seq[TablePartitionSpec]]("specs").flatMap(_.keySet).distinct
+        outputObjs += tablePrivileges(table, cols)
+
+      case "AlterTableRenameCommand" =>
+        val oldTable = getPlanField[TableIdentifier]("oldName")
+        val newTable = getPlanField[TableIdentifier]("newName")
+        outputObjs += tablePrivileges(oldTable, actionType = PrivilegeObjectActionType.DELETE)
+        outputObjs += tablePrivileges(newTable)
+
+      case "AlterTableRenamePartitionCommand" =>
+        val table = getTableName
+        val cols = getPlanField[TablePartitionSpec]("oldPartition").keySet.toSeq
+        outputObjs += tablePrivileges(table, cols)
+
+      case "AlterTableSerDePropertiesCommand" =>
+        val table = getTableName
+        val cols = getPlanField[Option[TablePartitionSpec]]("partSpec")
+          .toSeq.flatMap(_.keySet)
+        outputObjs += tablePrivileges(table, cols)
+
+      case "AlterTableSetLocationCommand" =>
+        val table = getTableName
+        val cols = getPlanField[Option[TablePartitionSpec]]("partitionSpec")
+          .toSeq.flatMap(_.keySet)
+        outputObjs += tablePrivileges(table, cols)
+
+      case "AlterTableSetPropertiesCommand" |
+          "AlterTableUnsetPropertiesCommand" =>
+        val table = getTableName
+        outputObjs += tablePrivileges(table)
+
+      case "AlterViewAsCommand" =>
+        val view = getPlanField[TableIdentifier]("name")
+        outputObjs += tablePrivileges(view)
+        buildQuery(getQuery, inputObjs)
+
+      case "AlterViewAs" =>
+
+      case "AnalyzeColumnCommand" =>
+        val table = getTableIdent
+        val cols = getPlanField[Option[Seq[String]]]("columnNames").getOrElse(Nil)
+        inputObjs += tablePrivileges(table, cols)
+
+      case "AnalyzePartitionCommand" =>
+        val table = getTableIdent
+        val cols = getPlanField[Map[String, Option[String]]]("partitionSpec")
+          .keySet.toSeq
+        inputObjs += tablePrivileges(table, cols)
+
+      case "AnalyzeTableCommand" |
+          "RefreshTableCommand" |
+          "RefreshTable" =>
+        inputObjs += tablePrivileges(getTableIdent)
+
+      case "AnalyzeTablesCommand" |
+          "ShowTablesCommand" =>
+        val db = getPlanField[Option[String]]("databaseName")
+        if (db.nonEmpty) {
+          inputObjs += databasePrivileges(db.get)
+        }
+
+      case "CacheTable" =>
+        // >= 3.2
+        outputObjs += tablePrivileges(getMultipartIdentifier)
+        val query = getPlanField[LogicalPlan]("table") // table to cache
+        buildQuery(query, inputObjs)
+
+      case "CacheTableCommand" =>
+        if (majorVersion == 3 && minorVersion == 1) {
```

Comment:
Implement different logic for different versions, looks better?

Reply:
Do you mean `case-match` with SPARK_VERSION?

Reply:
yes

Reply:
Maybe we can have a base implementation as a base template. For special commands which need to distinguish different spark versions, give a special implementation? It seems there are few commands that match the condition.